src/lib/coapcore/src/ace.rs

```diff
@@ -67,14 +69,45 @@ pub struct HeaderMap<'a> {
 
 impl HeaderMap<'_> {
     /// Merge two header maps, using the latter's value in case of conflict.
-    fn updated_with(self, other: Self) -> Self {
+    fn updated_with(&self, other: Self) -> Self {
```

Comment:
It's just an module-internal function; its users changed, and now this is more convenient.